src/main/java/com/iyzipay/Base64FileHelper.java

```diff
@@ -0,0 +1,37 @@
+package com.iyzipay;
+
+import com.iyzipay.exception.ImageFileEncodingException;
+
+import javax.xml.bind.DatatypeConverter;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+
+public final class Base64FileHelper {
+
+    private Base64FileHelper() {
+    }
+
+    public static String encode(String filePath) throws ImageFileEncodingException {
+        FileInputStream fileInputStream = null;
+        byte[] bytesArray;
+
+        try {
+            File file = new File(filePath);
+            bytesArray = new byte[(int) file.length()];
+            fileInputStream = new FileInputStream(file);
+            fileInputStream.read(bytesArray);
+        } catch (IOException e) {
+            throw new ImageFileEncodingException("Image file cannot be base64 encoded", e);
+        } finally {
+            if (fileInputStream != null) {
+                try {
+                    fileInputStream.close();
+                } catch (IOException ignored) {
+                }
+            }
+        }
+
+        return bytesArray != null ? DatatypeConverter.printBase64Binary(bytesArray) : null;
```

Comment:
**bytesArray** değişkeni sadece `new File(filePath)` kısmında hata olduğunda **null** olabilir. O kısımda da sadece **NullPointerException** fırlatılıyor. Dolayısıyla NPE'yi yakalamadığımız için zaten bu satır çalışmayacak.
Sonuç olarak buradaki null kontrolü kaldırılabilir.

Reply:
degistirdim.